File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        int AAA = 0, BBB = 0;
        for (int i = 1; i < colors.length() - 1; ++i) {
            if (colors.charAt(i) == colors.charAt(i - 1) && colors.charAt(i) == colors.charAt(i + 1)) 
                if (colors.charAt(i) == 'A')
                    ++AAA;
                else 
                    ++BBB;
        }
        return AAA > BBB;
    }
}

File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int AAA = 0, BBB = 0;
        for (int i = 1; i < size(colors) - 1; ++i) {
            if (colors[i] == colors[i - 1] and colors[i] == colors[i + 1]) 
                if (colors[i] == 'A')
                    ++AAA;
                else 
                    ++BBB;
        }
        return AAA > BBB;
    }
};